src/rgw/rgw_rados.cc

```diff
@@ -5071,12 +5080,30 @@ int RGWRados::create_pool(rgw_bucket& bucket)
   if (ret < 0)
     return ret;
 
+  if (! bucket.name_space.empty()) {
+    librados::IoCtx io_ctx;
+    ret = rad->ioctx_create(pool.c_str(), io_ctx);
+    if (ret) {
+      return ret;
+    }
+    io_ctx.set_namespace(bucket.name_space);
```

Comment:
the `set_namespace()` call just tells the `librados::IoCtx` object which namespace to use for subsequent rados requests. so this will just initialize an IoCtx and then throw it away

instead, we'll need to go through and find everywhere that we're calling `ioctx_create()` for one of these pools, and add the corresponding calls to `set_namespace()`. there are quite a few, so we should probably refactor those into  a common helper function so we don't miss any. `RGWRados::open_pool_ctx()` looks like a good candidate, if it could take the pool as an `rgw_bucket` instead of a `std::string`